yrb-to-javascript transpiler

ekashida opened the pull request:
@caridy @apipkin

ekashida committed:
initial commit of yala-to-javascript transpiler

caridy commented on lib/transpiler/index.js:
```diff
@@ -0,0 +1,12 @@
+var parse = require('./lib/parser').parse;
```
move this file into `lib/transpiler/yala/index.js`

caridy commented on lib/transpiler/index.js:
```diff
@@ -0,0 +1,12 @@
+        return new Error('Only strings can be transpiled');
+    }
+
+    return {
```
let's just return the `parse(message)`

caridy commented on lib/transpiler/lib/lexer.js:
```diff
@@ -0,0 +1,54 @@
+module.exports.tokenize = function (str, trim) {
```
let's do: `module.exports= function (str, trim) {` to export the function directly without any specific member.

caridy commented on lib/transpiler/lib/parser.js:
```diff
@@ -0,0 +1,121 @@
+    }
+];
+
+module.exports.parse = function parse (str) {
```
expose the function directly, same as above

caridy commented on lib/transpiler/example/basic.js:
```diff
@@ -0,0 +1,17 @@
+var transpile = require('../').transpile,
```
move this into `transpiler/yala/README.md` or into `examples`

ekashida committed:
export the tokenize function directly

ekashida committed:
export the parse function directly

ekashida committed:
account for multiple transpiler support

ekashida committed:
add support for base case formatters

Since we format certain strings using bracket-notation (e.g. # => ${#}),
we want to apply these transformations at the very end when we are done
parsing the message. Base case formatters hook into this moment and
allow us to format strings as they leave the parser.

ekashida committed:
`#` chars in plural format messages should be output as `${#}`

ekashida committed:
remove unused plain-string formatter

ekashida committed:
better names and more comments

ekashida committed:
add some validation

- options must come in name/pattern pairs
- options must include the default 'other' option

ekashida committed:
bad transpiler input should throw

ekashida committed:
api documentation

ekashida committed:
initial commit of readme

ekashida committed:
export the transpile method directly

ekashida committed:
add check for brackets that weren't closed

ekashida committed:
initial commit of yala transpiler unit tests

ekashida committed:
add license headers

ekashida committed:
add support for time/date/number format types

caridy commented on lib/transpilers/yala/index.js:
```diff
@@ -0,0 +1,25 @@
+        throw new Error('Only strings can be transpiled');
+    }
+
+    return {
```
`return parse(pattern)` should be just enough.

caridy commented:
this is looking great. let's wait until we can validate that it works fine with the stuff that tony is doing, then we can merge it.

also, the part where the plugin selects the transpiler, and call it to transform the tokens is also missing. transpiler should be a configuration option (just like format is today), so in the future we can support other transpilers.

ekashida committed:
return only the parsed object directly

caridy committed:
adding transpiler argument

ekashida committed:
Merge pull request #1 from caridy/yala

adding transpiler configuration option

ekashida committed:
rename from yala to yrb

caridy commented:
+1
